UnitySetup/UnitySetup.psm1

```diff
@@ -61,47 +64,55 @@ class UnitySetupInstance {
                 }
             }
             ([OperatingSystem]::Linux) {
                 $this.Components = [UnitySetupComponent]::Linux
-
-                throw "UnitySetupInstance has not been implemented on the Linux platform. Contributions welcomed!";
+                $playbackEnginePath = Join-Path -Path "$Path" -ChildPath "Editor/Data/PlaybackEngines" -Resolve;
+                @{
+                    [UnitySetupComponent]::Documentation  = , (Join-Path -Path "$Path" -ChildPath "Documentation" -Resolve);
+                    [UnitySetupComponent]::StandardAssets = , (Join-Path -Path "$Path" -ChildPath "Standard Assets" -Resolve);
+                    [UnitySetupComponent]::Mac            = , (Join-Path -Path "$playbackEnginePath" -ChildPath "MacStandaloneSupport/Variations/macos_x64_player_development_mono" -Resolve);
+                    [UnitySetupComponent]::Mac_IL2CPP     = , (Join-Path -Path "$playbackEnginePath" -ChildPath "MacStandaloneSupport/Variations/macos_x64_player_development_il2cpp" -Resolve);
+                    [UnitySetupComponent]::Windows        = , (Join-Path -Path "$playbackEnginePath" -ChildPath "WindowsStandaloneSupport" -Resolve);
+                    [UnitySetupComponent]::Linux_IL2CPP   = , (Join-Path -Path "$playbackEnginePath" -ChildPath "LinuxStandaloneSupport/Variations/linux64_player_development_il2cpp" -Resolve);
+                    [UnitySetupComponent]::Linux_Server   = , (Join-Path -Path "$playbackEnginePath" -ChildPath "LinuxStandaloneSupport/Variations/linux64_server_development_mono" -Resolve);
+                }
             }
             ([OperatingSystem]::Mac) {
                 $this.Components = [UnitySetupComponent]::Mac
-                $playbackEnginePath = [io.path]::Combine("$Path", "PlaybackEngines");
+                $playbackEnginePath = Join-Path -Path "$Path" -ChildPath "PlaybackEngines" -Resolve;
                 @{
-                    [UnitySetupComponent]::Documentation  = , [io.path]::Combine("$Path", "Documentation");
-                    [UnitySetupComponent]::StandardAssets = , [io.path]::Combine("$Path", "Standard Assets");
-                    [UnitySetupComponent]::Mac_IL2CPP     = , [io.path]::Combine("$playbackEnginePath", "MacStandaloneSupport/Variations/macosx64_development_il2cpp");
-                    [UnitySetupComponent]::Windows        = , [io.path]::Combine("$playbackEnginePath", "WindowsStandaloneSupport");
-                    [UnitySetupComponent]::Linux          = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport/Variations/linux64_headless_development_mono");
-                    [UnitySetupComponent]::Linux_IL2CPP   = , [io.path]::Combine("$playbackEnginePath", "LinuxStandaloneSupport/Variations/linux64_headless_development_il2cpp");
+                    [UnitySetupComponent]::Documentation  = , (Join-Path -Path "$Path" -ChildPath "Documentation" -Resolve);
```

Comment:
Unfortunately the `-Resolve` flag will cause an error to be thrown if the path doesn't exist, you'll need to remove those.

Reply:
Nit: Feel free to drop the `-Path` and `-ChildPath` for brevity if you want to.